### ISSUE TITLE
[论文复现] ConvS2S中缺个conv_tbc的算子，用conv1d后速度慢了五六倍

### ISSUE THREAD

**Issue by jiaohuix** [PaddlePaddle/Paddle#35257]
- 标题：ConvS2S中缺个conv_tbc的算子，用conv1d后速度慢了五六倍
- 版本、环境信息：
    1）PaddlePaddle版本：2.1.2
    2）CPU：i5
    3）GPU：3060 cuda11.2 cudnn8.0
    4）系统环境：windows10，Python版本3.6.8
- 模型信息
    1）模型名称 ConvS2S
    2）使用数据集名称 WMT'14 English to German 
    3）模型链接 https://github.com/pytorch/fairseq
- 复现信息：把fairseq/fairseq/modules/conv_tbc.py中conv_tbc函数return换成
-        return F.conv1d(input.permute(1, 2, 0), self.weight, self.bias, padding=self.padding[0]).permute(2, 0, 1)
- 问题描述：没有对应算子，用conv1d替代会慢五六倍（decoder）
- fair使用conv_tbc:
![1](https://user-images.githubusercontent.com/68314259/131287850-553f47d4-3ce6-4c9d-891c-7904ba4ae40f.png)

-fair中用conv1d
![2](https://user-images.githubusercontent.com/68314259/131287862-f36d1e9f-6a43-41b9-a235-5b5bc496d911.png)

**Comment by jiaohuix** [comment]
还有把weights变成：
        self.weight = self.create_parameter(shape=[out_channels, in_channels,self.kernel_size[0]])

**Comment by TCChenlong** [comment]
补充一个case：
```
import torch
import time
import paddle
import paddle.nn.functional as F

x = torch.ones([1000, 64, 3])
weight = torch.nn.Parameter(
            torch.Tensor(3, 3, 64)
        )
bias = torch.nn.Parameter(torch.Tensor(64))

start = time.time()
conv = torch.conv_tbc(x, weight, bias)
end = time.time()
print('conv_tbc time:' + str(end - start))


x = paddle.to_tensor(x.numpy())
weight = paddle.to_tensor(weight.detach().numpy())
bias = paddle.to_tensor(bias.detach().numpy())

total_start = time.time()
x = x.transpose((1, 2, 0))
weight = weight.transpose((2, 0, 1))
conv_start = time.time()
conv = F.conv1d(x, weight, bias)
conv_end = time.time()
conv = conv.transpose((2, 0, 1))
total_end = time.time()
print('conv1d time: ' + str(conv_end - conv_start))
print('total time: ' + str(total_end - total_start))
```
output:
```
conv_tbc time:0.01158595085144043
conv1d time: 0.01240086555480957
total time: 0.01693892478942871
```

**Comment by TCChenlong** [comment]
可以用 conv1d 绕过，且速度高于竞品
![image](https://user-images.githubusercontent.com/11935832/131623112-b24b5918-ac8a-4896-ab8e-43581e19d17a.png)
![image](https://user-images.githubusercontent.com/11935832/131623131-150776f2-d068-41cd-9a54-c7dc6bd1e562.png)

**Comment by TCChenlong** [comment]
但是，如果 shape 改为  [512, 64, 512]，则时间变为
![6c13c5e9d064225111524cba619af517](https://user-images.githubusercontent.com/11935832/131626682-ed0e1648-af8f-43ee-8400-6252f4a13275.png)
